minetest/minetest
# Proselytize the network. Use IEEE F32

**This is a bit messy, here's an explanation:**

* `NetworkPacket` reading and writing changed to F32
    * See changes in `networkpacket.cpp`. That's all.
* `SimpleSoundSpec` unified de-/serialization
    * Moved functions to `sound.h`
    * Introduced `CONTENTFEATURES_VERSION` as reference for new parameters
* `visual_size` changed to `v3f` (from previously 2D-vector)
    * Unrelated to the network changes but a welcome chance to change the size of this parameter.

Many serialization versions were raised, for safety and an easier future. The client would already abort when it receives unsupported ContentFeatures data, but bumping the versions for the other serialization methods makes git blame easier to use.

**How to test:**
1) Start a world with an old client; add boats, carts, mobs and check whether it looks identical with this PR
2) Install a mobs mod, shooter and a player to fight with them. This will confirm that following code areas work (which were changed):
 * Object sending: Positions, animations, speed
 * Sounds: gain, maybe also fade and pitch (dig nodes too, to ensure the right sounds are played)
 * Particles: (use grenades or blood) If they show up for both players, then they work correctly

**What was not changed:**
Anything related to the map saving. Mobs which were added in a world using this PR **must** show up the same way in the current HEAD build afterwards.
